### PR TITLE
assignment for module_2

### PR DESCRIPTION
**Pull request by fushen16**
create a date utility using revealing pattern

### CODE REVIEW

File: dateUtility.js [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+    if(!_privateDate) {
+      _initDate();
+    }
+
+    return _privateDate.toLocaleString(locale, { month: 'long' });
+  }
+
+  //getDate() and getDate('milliseconds') return milliseconds
+  //getDate('formatted') return date in the form of 'Month Day, Year'
+  function getDate(format) {
```
Comment (jamdon) [inline dateUtility.js]: The `format` parameter here should be an object with a format property on it. Usage would look like:

`dateUtility.getDate({ format: "milliseconds" });`

**Comment by jamdon** [comment]
Everything works great with one small exception. The specification for getDate is to take an object as a parameter - see inline comment. Additionally, if nothing is passed in, you want to check if the parameter is undefined, not null. Let me know if you want it edited as-is or I'll review it again after updates if you want.

**Comment by fushen16** [comment]
I see. Thank you, James. I can make the change. After change, should I do
the pull request again?

Fulin

On Tue, Apr 26, 2016 at 11:41 PM, jamdon notifications@github.com wrote:

> Everything works great with one small exception. The specification for
> getDate is to take an object as a parameter - see inline comment.
> Additionally, if nothing is passed in, you want to check if the parameter
> is undefined, not null.
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> https://github.com/JSCRIPT300-Spring2016/Module_2/pull/16#issuecomment-214985741

## 

Fulin Shen
541-357-8618

**Comment by jamdon** [comment]
Works great! 👍

**Pull request closed without merging**